homeassistant/components/home_connect/config_flow.py

```diff
@@ -50,4 +51,10 @@ async def async_oauth_create_entry(self, data: dict) -> ConfigFlowResult:
                 self._get_reauth_entry(),
                 data_updates=data,
             )
+        await self.async_set_unique_id(
+            jwt.decode(
+                data["token"]["access_token"], options={"verify_signature": False}
+            )["sub"]
+        )
+        self._abort_if_unique_id_configured()
```

Comment:
We should do a migration to get the existing entries to set their unique id as well

Reply:
We should also make sure that with a reauth, we get the same user id back, because if users want to change accounts, they should create a new entry